Wox.Core/Plugin/PluginManager.cs

```diff
@@ -92,21 +93,36 @@ public static void LoadPlugins(PluginsSettings settings)
             Settings.UpdatePluginSettings(_metadatas);
             AllPlugins = PluginsLoader.Plugins(_metadatas, Settings);
         }
+
+        /// <summary>
+        /// Call initialize for all plugins
+        /// </summary>
+        /// <returns>return the list of failed to init plugins or null for none</returns>
         public static void InitializePlugins(IPublicAPI api)
         {
             API = api;
+            var failedPlugins = new ConcurrentQueue<PluginPair>();
             Parallel.ForEach(AllPlugins, pair =>
             {
-                var milliseconds = Stopwatch.Debug($"|PluginManager.InitializePlugins|Init method time cost for <{pair.Metadata.Name}>", () =>
+                try
                 {
-                    pair.Plugin.Init(new PluginInitContext
+                    var milliseconds = Stopwatch.Debug($"|PluginManager.InitializePlugins|Init method time cost for <{pair.Metadata.Name}>", () =>
                     {
-                        CurrentPluginMetadata = pair.Metadata,
-                        API = API
+                        pair.Plugin.Init(new PluginInitContext
+                        {
+                            CurrentPluginMetadata = pair.Metadata,
+                            API = API
+                        });
                     });
-                });
-                pair.Metadata.InitTime += milliseconds;
-                Log.Info($"|PluginManager.InitializePlugins|Total init cost for <{pair.Metadata.Name}> is <{pair.Metadata.InitTime}ms>");
+                    pair.Metadata.InitTime += milliseconds;
+                    Log.Info($"|PluginManager.InitializePlugins|Total init cost for <{pair.Metadata.Name}> is <{pair.Metadata.InitTime}ms>");
+                }
+                catch (Exception e)
+                {
+                    Log.Exception(nameof(PluginManager), $"Fail to Init plugin: {pair.Metadata.Name}", e);
+                    pair.Metadata.Disabled = true; // TODO: not sure this really disable it later on
```

Comment:
Please check back on this, not sure if it's possible all the time.

Reply:
The code seems like a complete mess when looking at where the data of the plugins is kept, it is sometimes from the plugin manager and sometimes PluginsSettings hold some of the data.